### ISSUE TITLE
add option to disable KAYAK and social media platforms on a per platform basis in settings

### ISSUE THREAD

**Issue by foss-** [organicmaps/organicmaps#7342]
⚠ Have you searched for similar, already existing issues?

Yes.

**Is your feature request related to a problem? Please describe.**
App quality has decreased already since the KAYAK situation began:

- F-Droid lists an outdated version of Organic Maps with outdated map data from september 2023, while the february 2024 release has been published.
- on iOS some locations now show a prominent entry in the listing `K Photos, reviews, booking`

**Describe the ideal solution**
Not a KAYAK user and I won't become one, so information is occupying screenspace for which I have no use.
Please allow users to disable KAYAK entirely and while at it, why not allow disabling social services on a per platform level as well? I never will visit at tiktok or facebook like, so getting rid of those would be very useful.

**Describe alternatives you have considered**
An alternative could be to make KAYAK opt-in so any avid kayak user can enable the option and book away their hotels.

**Additional context**
There already is a PR which would resolve some of the mentioned issues: https://github.com/organicmaps/organicmaps/pull/6720#issuecomment-1916865558

Sadly there was a reply (yet?) as to why that PR was closed without merge.

**Comment by RedAuburn** [comment]
i think it shouldn't be too hard to implement, but the biggest issue is bloating the settings. maybe we can just add debug commands (eg. `?no-facebook-links`). then the small amount of users that want this can do it, but it doesn't complicate things for the average user

**Comment by Jean-BaptisteC** [comment]
Add one setting to disable all links can be an option

**Comment by foss-** [comment]
@fitojb noticing a lot of downvotes from you around the kayak topic. Can you share your reasoning and thoughts? Why do you think adding settings toggles is not a good idea?

**Comment by pastk** [comment]
Making a separate setting for each social platform / 3rd party service will bloat the settings too much IMHO.
Adding one or two settings (one for social links coming from OSM, another for integration with commercial services) I think is totally fine.

Actually per-platform settings could be implemented in a non-bloating way too - see how the "Routing options" opens a separate screen with 4 detailed settings. But its more work to implement and then maintain this code.

**Comment by Bubble68** [comment]
> Making a separate setting for each social platform / 3rd party service will bloat the settings too much IMHO.

Yes. It should be one setting to disable all external links. I don’t know why developers don’t want to do this. Stubbornness is the quality of donkeys.

**Comment by RedAuburn** [comment]
> Yes. It should be one setting to disable all external links. I don’t know why developers don’t want to do this. Stubbornness is the quality of donkeys.

You're free to submit a PR...

**Comment by Kassagi** [comment]
> PR already exists

Are you talking about this?
https://github.com/organicmaps/organicmaps/pull/6720

> Exclude contact information from hiding

Why? Let’s create toggle to blur addresses of corporations 😂

**Comment by RedAuburn** [comment]
> Maybe in F-Droid we should add open source solutions instead? Or completely remove that link. AKA light version.

there is no open source solution, if there was it would have been used in the first place

**Comment by kari909** [comment]
I don’t find Kayak link useful at all. There is https://github.com/organicmaps/organicmaps/pull/6720 PR to fix situation, but it is closed. Many people agree to remove it only from privacy oriented people version, but it is not the case for some why for devs…

**Comment by RedAuburn** [comment]
i guess we can do a poll, but IMO the real issue here is that the social media sections take up too much space. Disabling social medias is niche and too OSMAnd-like imo.
